internal/monitors/statsd/parser.go

```diff
@@ -0,0 +1,153 @@
+package statsd
+
+import (
+	"strings"
+)
+
+type fieldPattern struct {
+	substrs        []string
+	startWithField bool
+}
+
+type converter struct {
+	pattern    *fieldPattern
+	metric     *fieldPattern
+	dimensions []string
+}
+
+func initConverter(input *converterInput) *converter {
+	if input.Pattern == "" {
+		logger.Error("[pattern] is required for a converter")
```

Comment:
It is generally better to do validation of this stuff in the `Config.Validate() error` method and then these functions can be more focused and not have to deal with error checking.